Sources/Commands/PackageTools/InstalledPackages.swift

```diff
@@ -0,0 +1,122 @@
+        var product: String?
+        
+        func run(_ tool: SwiftTool) async throws {
+            let swiftpmBinDir = try tool.fileSystem.getOrCreateSwiftPMInstalledBinariesDirectory()
+            
+            let env = ProcessInfo.processInfo.environment
+            
+            if let path = env.path, !path.contains(swiftpmBinDir.pathString), !globalOptions.logging.quiet {
+                tool.observabilityScope.emit(warning: "PATH doesn't include \(swiftpmBinDir.pathString)! This means you won't be able to access the installed executables by default, and will need to specify the full path.")
+            }
+            
+            let installedPackagesJSONPath = try tool.fileSystem.dotSwiftPM.appending(component: "installedPackages.json")
+            
+            var alreadyExisting = (try? InstalledPackage.registered(registryPath: .init(installedPackagesJSONPath), withFS: tool.fileSystem)) ?? []
+            
+            let workspace = try tool.getActiveWorkspace()
+            let packageRoot = try tool.getPackageRoot()
+
+            let packageGraph = try await workspace.loadRootPackage(at: packageRoot, observabilityScope: tool.observabilityScope)
+            
+            let possibleCanidates = packageGraph.products.filter { $0.type == .executable }
+            let productToInstall: Product
+            
+            switch possibleCanidates.count {
+            case 0:
+                throw StringError("No Executable Products in Package.swift.")
+            case 1:
+                productToInstall = possibleCanidates[0]
+            default: // More than one, check for possible
+                guard let product, let first = possibleCanidates.first(where: { $0.name == product }) else {
+                    throw StringError("Multiple canidates found, however, no product was specified. specify a product with the --product")
+                }
+                
+                productToInstall = first
+            }
+            
+            if let existingPkg = alreadyExisting.first(where: { $0.name == productToInstall.name }) {
+                throw StringError("\(productToInstall.name) is already installed at \(existingPkg.url)")
+            }
+            
+            try tool.createBuildSystem(explicitProduct: productToInstall.name).build(subset: .product(productToInstall.name))
+            
+            let binPath = try tool.buildParameters().buildPath.appending(component: productToInstall.name)
+            let finalBinPath = swiftpmBinDir.appending(component: binPath.basename)
+            try tool.fileSystem.copy(from: binPath, to: finalBinPath)
+            
+            let pkgInstance = InstalledPackage(name: productToInstall.name, url: .init(finalBinPath))
+            alreadyExisting.append(pkgInstance)
+            
+            try JSONEncoder().encode(path: installedPackagesJSONPath, fileSystem: tool.fileSystem, alreadyExisting)
+        }
+    }
+    
+    struct Remove: SwiftCommand {
```

Comment:
IMO `Uninstall` makes it a more obvious and discoverable counterpart to `Install`.

```suggestion
    struct Uninstall: SwiftCommand {
```